runtime/gc_base/RootScanner.cpp

```diff
@@ -930,10 +936,30 @@ MM_RootScanner::scanDoubleMappedObjects(MM_EnvironmentBase *env)
 				}
 			}
 		}
-		reportScanningEnded(RootScannerEntity_DoubleMappedObjects);
+		reportScanningEnded(RootScannerEntity_DoubleMappedOrVirtualLargeObjectHeapObjects);
 	}
 }
-#endif /* J9VM_GC_ENABLE_DOUBLE_MAP */
+#endif /* defined(J9VM_GC_ENABLE_DOUBLE_MAP) */
+
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+void
+MM_RootScanner::scanObjectsInVirtualLargeObjectHeap(MM_EnvironmentBase *env)
+{
+	if (_singleThread || J9MODRON_HANDLE_NEXT_WORK_UNIT(env)) {
+		GC_HeapRegionIteratorVLHGC regionIterator(_extensions->heap->getHeapRegionManager());
+		MM_HeapRegionDescriptorVLHGC *region = NULL;
+		reportScanningStarted(RootScannerEntity_DoubleMappedOrVirtualLargeObjectHeapObjects);
+		while (NULL != (region = regionIterator.nextRegion())) {
```

Comment:
We can probably do better, by finding offheap objects by iterating the map (dataAddr to Object) in Sparse Heap.